### PR TITLE
[MS] Fixed file/user list item background and options button

### PR DESCRIPTION
**Pull request by Max-7**
Two very small changes related to the file list item:
1. correctly displays a background when selected
2. hide the options button when item not hover for files and users

- [X] Keep changes in the pull request as small as possible
- [X] Ensure the commit history is sanitized
- [X] Give a meaningful title to your PR
- [X] Describe your changes

### CODE REVIEW

**Comment by fabienSvtr** [comment]
In `FileListItem.vue` we should remove the line 161 `:focus` because it creates a bug when we maintain left click and drag on a not selected item.

**Review by fabienSvtr — APPROVED** [review]

File: client/src/components/files/FileCard.vue [hunk 1]
```diff
@@ -4,10 +4,18 @@
...
     class="card"
     @click="$emit('click', $event, file)"
+    @mouseenter="isHovered = true"
+    @mouseleave="isHovered = false"
   >
     <div
       class="card-option"
-      @click.stop="$emit('menuClick', $event, file)"
+      v-show="isHovered || menuOpened"
+      @click.stop="
```
Comment (NicoTuxx) [inline client/src/components/files/FileCard.vue:13]: all the following code makes reading the template laborious, should be moved to a `onSomethingClick` function.

File: client/src/components/files/FileListItem.vue [hunk 2]
```diff
@@ -72,8 +73,14 @@
     <div class="file-options ion-item-child-clickable">
       <ion-button
         fill="clear"
+        v-show="isHovered || menuOpened"
         class="options-button"
-        @click.stop="$emit('menuClick', $event, file)"
+        @click.stop="
+          menuOpened = true;
```
Comment (NicoTuxx) [inline client/src/components/files/FileListItem.vue:79]: same as previous comment.

File: client/src/components/users/UserCard.vue [hunk 3]
```diff
@@ -21,8 +21,13 @@
     </div>
     <div
       class="card-option"
-      v-show="showOptions"
-      @click.stop="$emit('menuClick', $event, user)"
+      v-show="isHovered || menuOpened"
+      @click.stop="
+        menuOpened = true;
```
Comment (NicoTuxx) [inline client/src/components/users/UserCard.vue:26]: same here too.

File: client/src/components/users/UserListItem.vue [hunk 4]
```diff
@@ -55,10 +55,15 @@
     <!-- options -->
     <div class="user-options ion-item-child-clickable">
       <ion-button
-        v-show="showOptions"
+        v-show="(isHovered || menuOpened) && !hideOptions"
         fill="clear"
         class="options-button"
-        @click.stop="$emit('menuClick', $event, user)"
+        @click.stop="
+          menuOpened = true;
```
Comment (NicoTuxx) [inline client/src/components/users/UserListItem.vue:62]: same.

**Review by NicoTuxx — CHANGES_REQUESTED** [review]

**Review by NicoTuxx — CHANGES_REQUESTED** [review]
A weird behavior is happening when selecting elements and then clicking on an other one.
It only seems to happen when we click on the element himself, not his options button.
It does not happen when we click outside of the list too.
https://github.com/Scille/parsec-cloud/assets/19772607/46f8aa08-e1d6-48c0-b1d0-cb0e3c80d827

**Comment by Max-7** [comment]
> A weird behavior is happening when selecting elements and then clicking on an other one. It only seems to happen when we click on the element himself, not his options button. It does not happen when we click outside of the list too. https://github.com/Scille/parsec-cloud/assets/19772607/46f8aa08-e1d6-48c0-b1d0-cb0e3c80d827

It's because of the mocks. You're travelling inside the folder `Dir1` but since the mocks always returns the same files/folders, it feels like the selection is just removed. If you look at the header, you'll see that you are now inside `/Dir1` instead of `/`.

**Review by NicoTuxx — APPROVED** [review]

**Pull request merged**